### ISSUE TITLE
#[deprecated] incorrectly warns in the same crate if derives are involved

### ISSUE THREAD

**Issue by gnzlbg** [rust-lang/rust#55417]
```rust

#[derive(Debug)]
#[deprecated(since = "0.1.1", note = "oh noes")]
pub struct A;

#[derive(PartialOrd)]
#[deprecated(since = "0.1.1", note = "oh noes")]
pub struct B;

fn main() {}

```

([Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2015&gist=7255596c2dc2dfdf3671bb17e98b5bc8))

Errors:

```
   Compiling playground v0.0.1 (/playground)
warning: use of deprecated item 'A': oh noes
  |
  = note: #[warn(deprecated)] on by default

warning: use of deprecated item 'B': oh noes

error[E0277]: can't compare `B` with `B`
 --> src/main.rs:6:10
  |
6 | #[derive(PartialOrd)]
  |          ^^^^^^^^^^ no implementation for `B == B`
  |
  = help: the trait `std::cmp::PartialEq` is not implemented for `B`

error: aborting due to previous error
```

An easy fix might be to `#[allow(deprecated)]` in the custom derive implementations provided by Rust. Right now one has to manually `#[allow(deprecated)]` on all deprecated structs with derives, which is a pain, and also on all their impls...

**Comment by joshlf** [comment]
Same as #38832?

**Comment by joshlf** [comment]
Should this really have been closed? The fix makes the error message better, but doesn't actually address the issue with the deprecation warning being emitted on the generated impl.

**Comment by petrochenkov** [comment]
https://github.com/rust-lang/rust/pull/65310 should indeed address the issue and remove the deprecation warning.
Perhaps it's not in nightly yet?

**Comment by joshlf** [comment]
Ah you're probably right. I'll check again tomorrow.
